include/zephyr/drivers/counter.h

```diff
@@ -113,6 +113,43 @@ extern "C" {
 
 /**@} */
 
+/**
+ * @anchor COUNTER_CAPTURE_FLAGS
+ * @name Counter capture flags
+ *
+ * @brief Used by @ref counter_capture_callback_set.
+ * @{
+ */
+
+/**
+ * @brief Capture rising edge of an external input signal
+ */
+ #define COUNTER_CAPTURE_RISING_EDGE BIT(0)
+
+/**
+ * @brief Capture falling edge of an external input signal
+ */
+ #define COUNTER_CAPTURE_FALLING_EDGE BIT(1)
+
+/**
+ * @brief Capture both falling and rising edge of an external input signal
+ */
+ #define COUNTER_CAPTURE_BOTH_EDGES (COUNTER_CAPTURE_FALLING_EDGE | COUNTER_CAPTURE_RISING_EDGE)
+
+ /**@} */
+
+ /** @brief Counter capture callback
+ *
+ * @param dev       Pointer to the device structure for the driver instance
+ * @param chan_id   Channel ID
+ * @param flags     Configuration flags (COUNTER_CAPTURE_*)
```

Comment:
Since there is an *anchor* for flags:
```suggestion
 * @param flags     Configuration flags (@ref COUNTER_CAPTURE_FLAGS)
```
Ditto for `counter_capture_callback_set()` inline description comment.